src/workloads/execution/SecondaryReadsGenny.yml

```diff
@@ -0,0 +1,252 @@
+              Document:
+                x: {^RandomInt: {min: *initial_nb_docs, max: 1000000000}}
+                #s: {^Repeat: {count: *document_size, fromGenerator: {^Choose: {from: ["x"]}}}} --> why does this not work?
+                s: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+
+- TemplateName: ReadsOnSecondaryTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: ReadsOnSecondary}}
+    Type: CrudActor
+    Threads: {^Parameter: {Name: "Threads", Default: -1}}
+    ClientName: Secondary
+    Database: *db
+    Phases:
+      OnlyActiveInPhases:
+        Active: {^Parameter: {Name: "ActivePhases", Default: [-1]}}
+        NopInPhasesUpTo: *max_phase
+        PhaseConfig:
+          Duration: *test_duration
+          Collection: Collection0
+          Operations:
+          - OperationName: find
+            OperationCommand:
+              Filter: { x: {$gte: {^RandomInt: {min: 0, max: *initial_nb_docs}}}} # this should be *initial_nb_docs - *read_limit but we don't seem to have and add/subtract generator and it doesn't really impact performance
+              Options:
+                Limit: *read_limit
+                BatchSize: 200
+                ReadPreference:
+                  ReadMode: secondaryPreferred
+
+- TemplateName: ReplicationLagTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: ReplicationLag}}
+    Type: CrudActor
+    Threads: 1
+    Database: *db
+    Phases:
+      OnlyActiveInPhases:
+        Active: {^Parameter: {Name: "ActivePhases", Default: [-1]}}
+        NopInPhasesUpTo: *max_phase
+        PhaseConfig:
+          Repeat: 1
+          Collection: Collection0
+          Operations:
+          - OperationName: insertOne
+            OperationCommand:
+              Document: {s: 1}
+              OperationOptions:
+                WriteConcern:
+                  Level: 3
+                  Journal: true
+
+- TemplateName: CleanUpTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: CleanUp}}
+    Type: RunCommand
+    Threads: 1
+    Phases:
+      OnlyActiveInPhases:
+        Active: {^Parameter: {Name: "ActivePhases", Default: [-1]}}
+        NopInPhasesUpTo: *max_phase
+        PhaseConfig:
+          Repeat: 1
+          Database: *db
+          Operations:
+          - OperationName: RunCommand
+            OperationCommand:
+              drop: Collection0
+
+Actors:
+- ActorFromTemplate:
+    TemplateName: LoadInitialDataTemplate
+    TemplateParameters:
+      Name: LoadInitialData
+      ActivePhases: [0, 5, 10]
+
+- ActorFromTemplate:
```

Comment:
Is this shorter and clearer for using a template? How long is it if you just flat instantiate the quiesce actor here? I'm suspicious of templates that are only used once.

Reply:
Good point, while working with the test I've also noticed that templates aren't that great for readability so removing them where they are not necessary makes sense.